Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForest.java

```diff
@@ -1051,7 +1103,25 @@ public T threadPoolSize(int threadPoolSize) {
         }
 
         public T compactEnabled(boolean compactEnabled) {
-            this.compactEnabled = compactEnabled;
+            if (!this.compactEnabled) {
```

Comment:
Did you mean to write `if (!floatEnabled)` here?

Reply:
yes. Fixed.

Reply:
If I'm reading this correctly, it means that the builder will let you set compact enabled to true once, but will silently ignore you if you try to change it after that. If we later change the default to having compact enabled be true, this would prevent a user from ever creating a non-compact forest.

My suggestion is that we don't add logic to the builder methods. Let's keep all config validation logic in the constructor. Then, if the user choose an inconsistent set of options we can either throw an exception or override them and print a warning to stderr.

Reply:
https://github.com/aws/random-cut-forest-by-aws/issues/82